xmodule/modulestore/store_utilities.py

```diff
@@ -104,3 +106,26 @@ def get_draft_subtree_roots(draft_nodes):
     for draft_node in draft_nodes:
         if draft_node.parent_url not in urls:
             yield draft_node
+
+
+def derived_key(courselike_source_key, block_key, parent):
```

Comment:
`parent` is fairly ambiguous - parent of the source or of the destination? UsageKey? XBlock? BlockKey? I'd prefer sticking with the name `new_parent_block_key` or even better `dest_parent: BlockKey` with a type hint.